build_tools/build_libtorch.sh

```diff
@@ -95,17 +103,19 @@ checkout_pytorch() {
   fi
   cd $PYTORCH_ROOT
   git fetch --all
-  git checkout ${PYTORCH_BRANCH}
+  git checkout origin/${PYTORCH_BRANCH}
   git submodule update --init --recursive
 }
 
 build_pytorch() {
-  BUILD_SHARED_VAR="ON"
+  cd $PYTORCH_ROOT
+  # Uncomment the next line if you want to iterate on source builds
+  python setup.py clean
   if [[ $LIBTORCH_VARIANT = *"static"* ]]; then
-    BUILD_SHARED_VAR="OFF"
+    BUILD_SHARED_LIBS=OFF USE_LIGHTWEIGHT_DISPATCH=ON STATIC_DISPATCH_BACKEND=ON BUILD_LITE_INTERPRETER=ON BUILD_TEST=OFF USE_GLOO=OFF USE_MPS=OFF USE_PYTORCH_QNNPACK=OFF USE_OPENMP=OFF  USE_OBSERVERS=OFF USE_KINETO=OFF USE_EIGEN_FOR_BLAS=OFF _GLIBCXX_USE_CXX11_ABI=${CXX_ABI} USE_NCCL=OFF INTERN_DISABLE_ONNX=OFF BUILD_PYTHONLESS=1 USE_CUDA=OFF USE_MKL=OFF USE_XNNPACK=OFF USE_DISTRIBUTED=OFF USE_BREAKPAD=OFF USE_MKLDNN=OFF USE_QNNPACK=OFF USE_NNPACK=OFF ONNX_ML=OFF CMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES} python setup.py build
+  else
+    BUILD_TEST=OFF USE_GLOO=OFF USE_MPS=OFF USE_PYTORCH_QNNPACK=OFF USE_OPENMP=OFF  USE_OBSERVERS=OFF USE_KINETO=OFF USE_EIGEN_FOR_BLAS=OFF _GLIBCXX_USE_CXX11_ABI=${CXX_ABI} USE_NCCL=OFF INTERN_DISABLE_ONNX=OFF USE_CUDA=OFF USE_MKL=OFF USE_XNNPACK=OFF USE_DISTRIBUTED=OFF USE_BREAKPAD=OFF USE_MKLDNN=OFF USE_QNNPACK=OFF USE_NNPACK=OFF ONNX_ML=OFF USE_MKL=OFF CMAKE_OSX_ARCHITECTURES=${CMAKE_OSX_ARCHITECTURES} python setup.py  bdist_wheel -d $WHEELHOUSE
```

Comment:
The static case was defaulting to the static libtorch build. The bdist_wheel is what is used to build the shared libtorch and the wheel package. But let me see if bdist_wheel would build a static library (this deviates from upstream) since they only build the static library when they don't build wheels.

Reply:
Fair. It's not a problem. Just curious why there was a difference.

Reply:
I was able to get the same command to work. Will still test and validate the source and static builds some more and follow up with related commits.

Reply:
just for completeness I tested the .whl generated when building the static variant and it fails to dlopen the shared library which is expected. So if we want a usable .whl we build it in shared mode and in static mode only libtorch will be usable.